utils/src/main/kotlin/Extensions.kt

```diff
@@ -199,6 +200,18 @@ val NON_LINUX_LINE_BREAKS = Regex("\\r\\n?")
  */
 fun String.normalizeLineBreaks() = replace(NON_LINUX_LINE_BREAKS, "\n")
 
+/**
+ * Strip any user name off the URL represented by this [String]. Return the unmodified [String] if it does not represent
+ * a URL or if it does not include a user name.
+ */
+fun String.stripUserNameFromUrl() = try {
+    URI(this).let {
+        URI(it.scheme, null, it.host, it.port, it.rawPath, it.rawQuery, it.rawFragment).toString()
```

Comment:
Would it be more general to pass: 'authority' instead of 'host' + 'port' ? see also identities section

Why did you choose the raw* getters over the normal ones?

Reply:
1. I was simply using the constructor with the most arguments, to ensure nothing gets set to default values.

2. The raw getters (if available) do not decode anything, so they are the counterpart to setting the different fields.

Reply:
sorry, haven't mentioned java doc in my above comment. There is a section about constructors which examplify identical constructions of URIs. Using authority seems to be more general than using port and host in that sense that it not limiting it to hierarchical URIs

Reply:
If you're referring to the "in all cases" example in the "Identities" section of https://docs.oracle.com/javase/7/docs/api/java/net/URI.html, then I actually believe it's a mistake because if you look further down, there is no constructor that takes `userInfo` *and* `authority`. As I explicitly need to set `userInfo` to `null` there is only one constructor that I can use.